lib/room.cpp

```diff
@@ -2368,14 +2370,40 @@ Room::Changes Room::processStateEvent(const RoomEvent& e)
                 oldStateEvent
                     ? static_cast<const RoomAliasesEvent*>(oldStateEvent)->aliases()
                     : QStringList();
-            connection()->updateRoomAliases(id(), ae.stateKey(),
-                                            previousAliases, ae.aliases());
+            connection()->updateRoomAliases(id(), previousAliases, ae.aliases());
```

Comment:
You may want to preserve the canonical alias here somehow - or instead (actually, better), just ignore `RoomAliasesEvent` since the MSC says it's no more a thing.